stacker_blueprints/eks.py

```diff
@@ -0,0 +1,427 @@
+                           "will be created if it's not provided.",
+            "default": ""
+        },
+        "Version": {
+            "type": str,
+            "description": "Kubernetes version",
+            "default": "",
+        },
+        "SecurityGroupIds": {
+            "type": str,
+            "description": "A comma separated list of security group ids for "
+                           "controlling ENI access from EKS to workers",
+        },
+        "SubnetIds": {
+            "type": str,
+            "description": "A comma separated list of subnet ids where you "
+                           "will launch your worker nodes",
+        }
+    }
+
+    # This creates an IAM role which EKS requires, as described here:
+    # https://docs.aws.amazon.com/eks/latest/userguide/getting-started.html#eks-create-cluster
+    def create_iam_role(self):
+        eks_service_role_id = "EksServiceRole"
+        t = self.template
+        t.add_resource(
+            Role(
+                eks_service_role_id,
+                AssumeRolePolicyDocument=Policy(
+                    Statement=[
+                        Statement(
+                            Effect=Allow,
+                            Action=[AssumeRole],
+                            Principal=Principal("Service",
+                                                ["eks.amazonaws.com"])
+                        )
+                    ]
+                ),
+                Path="/",
+                ManagedPolicyArns=[
+                    "arn:aws:iam::aws:policy/AmazonEKSClusterPolicy",
+                    "arn:aws:iam::aws:policy/AmazonEKSServicePolicy",
+                ]
+            )
+        )
+        return GetAtt(eks_service_role_id, "Arn")
+
+    def get_iam_role(self):
+        role_arn = self.get_variables()["ExistingRoleArn"]
+        if role_arn:
+            return role_arn
+        return self.create_iam_role()
+
+    def create_template(self):
+        t = self.template
+        role_arn = self.get_iam_role()
+        variables = self.get_variables()
+
+        args = {}
+        version = variables["Version"]
+        if version:
+            args["Version"] = version
+
+        # This is a fully qualified stacker name, prefixed with the namespace.
+        eks_name_tag = self.context.get_fqn(variables["Name"])
+
+        t.add_resource(
+            eks.Cluster(
+                "EksCluster",
+                Name=eks_name_tag,
+                RoleArn=role_arn,
+                ResourcesVpcConfig=eks.ResourcesVpcConfig(
+                    SecurityGroupIds=variables["SecurityGroupIds"].split(","),
+                    SubnetIds=variables["SubnetIds"].split(","),
+                ),
+                **args
+            )
+        )
+
+        # Output the ClusterName and RoleArn, which are useful as inputs for
+        # EKS worker nodes.
+        t.add_output(Output("ClusterName", Value=eks_name_tag))
```

Comment:
I'd add more outputs for the Cluster & Role.  Usually we create the outputs in the same method where the resources are created, just to keep code close together.  Might be difficult with the Role, though I think there are some good examples in the codebase already.

Reply:
I think this exports everything that's needed, you can get everything via GetAtt(clustername, thing). I did the Role here on purpose, so that I can pass it in or create it. I can make it output in two places - once just after creation, and once after reusing an existing one, but I chose to spread things out and not duplicate. Your call, which do you prefer?